### ISSUE TITLE
Alltools docker image

### ISSUE THREAD

**Issue by holiman** [openethereum/openethereum#9405]
On docker [hub](https://hub.docker.com/r/parity/parity/), there seems to be only `latest` and `stable` built with any regularity (aside from release-builds). Afaiu, they're built off [this](https://github.com/paritytech/parity-ethereum/blob/master/docker/hub/Dockerfile) dockerfile. 

I would be very greatful if you could add another docker image to docker hub, `alltools`, which also contains `parity-evm` (and any other binaries that the build process can generate). A similar thing exists for geth, with `ethereum/client-go:alltools-latest `

**Comment by holiman** [comment]
Here's an example dockerfile that contains `parity-evm`:

```

# Taken from https://github.com/paritytech/parity-ethereum/blob/master/docker/alpine/Dockerfile
FROM alpine:edge AS builder

# show backtraces
ENV RUST_BACKTRACE 1

RUN apk add --no-cache \
  build-base \
  cargo \
  cmake \
  eudev-dev \
  linux-headers \
  perl \
  rust git

RUN git clone https://github.com/paritytech/parity-ethereum /parity
WORKDIR /parity
RUN cargo build --release --target x86_64-alpine-linux-musl --verbose -p evmbin
RUN strip target/x86_64-alpine-linux-musl/release/parity-evm


FROM alpine:edge

# show backtraces
ENV RUST_BACKTRACE 1

RUN apk add --no-cache \
  libstdc++ \
  eudev-libs \
  libgcc

RUN addgroup -g 1000 parity \
  && adduser -u 1000 -G parity -s /bin/sh -D parity

USER parity
COPY --chown=parity:parity --from=builder /parity/target/x86_64-alpine-linux-musl/release/parity-evm ./

#ENTRYPOINT ["./parity"]

```

**Comment by 5chdn** [comment]
Actually, it's worth to add all of our tools to the parity docker image instead of maintaining multiple docker images.

**Comment by holiman** [comment]
Definitely! Anyone who wants a smaller image can just use that complete image as the basis of a multistage build, and pick out the cherries.

**Comment by holiman** [comment]
Any progress here  ? Would be really nice to have...

**Comment by sachincool** [comment]
@joshua-mir hey, Can I pick this up, I tried to do the same thing as said above in the example of `ethereum/client-go:alltools-latest` by simply Copying everything from the build `release/*` 

As mentioned above, We can make multiple Stages in the multi stage build where the source builder is same, making multiple Enviroment Versions super fast and easy eg:
```
$ docker build -t parity:release . --target=release
$ docker build -t parity:dev . --target=dev
$ docker images
REPOSITORY                                      TAG                 IMAGE ID            CREATED              SIZE
parity                                          dev                 1ac06e094d18        48 seconds ago       562MB
parity                                          release             5e702377839b        About a minute ago   16.4MB
```
I added the file in `scripts/docker/alpine/Dockerfile`

**Comment by holiman** [comment]
Hey ho OE Team! 
Could you please take a look at this? I suspect it would be pretty easy to fix this, once you've set up a pipeline for building/pushing docker images

**Comment by holiman** [comment]
Is this fixed?

**Comment by adria0** [comment]
Hi @holiman, we have been closing - to be honest in a very direct - way a lot of issues to check if they are still relevant for users.

@denisgranha, any input on this?

**Comment by denisgranha** [comment]
hey! we can add it when we add also another docker images for debian, etc

Not in the next days and I'll be on Holidays soon. 

End of August we could add some improvements on the docker images
